### ISSUE TITLE
[a11y]: Modal: Follow guidance on dismissing the modal by clicking elsewhere

### ISSUE THREAD

**Issue by tay1orjones** [carbon-design-system/carbon#13712]
Originally outlined in #12727:

> A modal dialog, by definition, is supposed to be the only operable thing on the screen. Theoretically, the area outside the modal should be _inert_. Clicking it should do nothing.
> Modals by design have a Cancel (or No) button as well as the X button to close, both of which allow affordances for a user to easily dismiss. There does not seem to be a good reason to allow the area outside the modal to be interactive.
> 
> The only [functional demonstration in the React storybook](https://react.carbondesignsystem.com/?path=/story/components-modal--with-state-manager) IS dismissable by clicking outside, even though Carbon says such [transactions modals](https://carbondesignsystem.com/components/modal/usage/#transactional-modal) are not dismissed by clicking outside.
> 
> The issue is made a bit problematic since Carbon also introduces the idea of a passive modal, which _can_ be dismissed in this way. In my [discussion on Modals](https://github.com/carbon-design-system/carbon/discussions/12726) I recommend that the Passive modal be called something else. There are lots of non-modal dialog prompts and interactions where clicking outside can dismiss content. I just feel that a modal dialog may not be the place to allow it.
> Note: I am trying to get clarity from ARIA authorities on whether clicking outside is even _allowed_ in a modal dialog. Their response may help inform the approach here. **Sev: discussion**
> 
> At the least, only the passive should be dismissable by clicking outside, to match the Carbon documentation.
